File: support_tools/swigwin-3.0.12/Examples/test-suite/python/complextest_runme.py
import complextest

a = complex(-1, 2)

if complextest.Conj(a) != a.conjugate():
    raise RuntimeError, "bad complex mapping"

if complextest.Conjf(a) != a.conjugate():
    raise RuntimeError, "bad complex mapping"


v = (complex(1, 2), complex(2, 3), complex(4, 3), 1)

try:
    complextest.Copy_h(v)
except:
    pass

File: support_tools/swigwin-3.0.12/Examples/test-suite/javascript/complextest_runme.js
var complextest = require("complextest");

a = [-1,2];

expected = [-1, -2];

a_c = complextest.Conj(a);
if (a_c.toString() != expected.toString())
  throw "Error in Conj(a)";

a_c_f = complextest.Conjf(a);
if (a_c_f.toString() != expected.toString())
    throw "Error in Conjf(a)";

v = new complextest.VectorStdCplx();
v.add([1,2]);
v.add([2,3]);
v.add([4,3]);
v.add(1);

// TODO: how to check validity?
complextest.Copy_h(v);
